clang/test/Preprocessor/predefined-arch-macros.c

```diff
@@ -4306,10 +4306,10 @@
 
 // Begin amdgcn tests ----------------
 
-// RUN: %clang -mcpu=gfx803 -E -dM %s -o - 2>&1 \
+// RUN: %clang -mcpu=gfx803 -E -dM -Xclang -fcuda-is-device %s -o - 2>&1 \
```

Comment:
C code compiled with target amdgcn should not depend on -fcuda-is-device since it is only compiled for device